Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,303 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import SnapKit
+import Shared
+import Storage
+import Deferred
+
+protocol ShareControllerDelegate: class {
+    func finish(afterDelay: TimeInterval)
+    func getValidExtensionContext() -> NSExtensionContext?
+    func getShareItem() -> Deferred<ShareItem?>
+}
+
+class ShareViewController: UIViewController {
+    private var shareItem: ShareItem?
+    private var separators = [UIView]()
+    private var actionRows = [UIView]()
+
+    private var stackView: UIStackView!
+    private var sendToDevice: SendToDevice?
+    weak var delegate: ShareControllerDelegate?
+
+    override var extensionContext: NSExtensionContext? {
+        get {
+            return delegate?.getValidExtensionContext()
+        }
+    }
+
+    private func makeSeparator() -> UIView {
+        let view = UIView()
+        view.backgroundColor = UX.separatorColor
+        separators.append(view)
+        return view
+    }
+
+    private func layoutSeparators() {
+        separators.forEach {
+            $0.snp.makeConstraints { make in
+                make.left.right.equalToSuperview()
+                make.height.equalTo(1)
+            }
+        }
+    }
+
+    private func makePageInfoRow() -> (row: UIView, pageTitleLabel: UILabel, urlLabel: UILabel) {
+        let row = UIView()
+
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.spacing = UX.pageInfoLineSpacing
+
+        row.addSubview(stackView)
+        stackView.snp.makeConstraints { make in
+            make.centerY.equalToSuperview()
+            make.left.right.equalToSuperview().inset(UX.pageInfoRowLeftInset)
```

Comment:
Using `leading` and `trailing` instead of using `left` `right` is recommended for LTR support